crates/nu-cli/src/config_files.rs

```diff
@@ -38,38 +41,95 @@ pub fn read_plugin_file(
-        let plug_path = plugin_filename.to_string();
-
-        if let Ok(contents) = std::fs::read(&plugin_path) {
-            perf(
-                &format!("read plugin file {}", &plug_path),
-                start_time,
-                file!(),
-                line!(),
-                column!(),
-                engine_state.get_config().use_ansi_coloring,
-            );
-            start_time = std::time::Instant::now();
-            eval_source(
-                engine_state,
-                stack,
-                &contents,
-                &plugin_filename,
-                PipelineData::empty(),
-                false,
-            );
-            perf(
-                &format!("eval_source plugin file {}", &plug_path),
-                start_time,
-                file!(),
-                line!(),
-                column!(),
-                engine_state.get_config().use_ansi_coloring,
+        // Open the plugin file
+        let mut file = match std::fs::File::open(&plugin_path) {
+            Ok(file) => file,
+            Err(err) => {
+                if err.kind() == std::io::ErrorKind::NotFound {
+                    log::warn!("Plugin file not found: {}", plugin_path.display());
+                } else {
+                    report_error_new(
+                        engine_state,
+                        &ShellError::GenericError {
+                            error: format!(
+                                "Error while opening plugin cache file: {}",
+                                plugin_path.display()
+                            ),
+                            msg: "plugin path defined here".into(),
+                            span,
+                            help: None,
+                            inner: vec![err.into()],
+                        },
+                    );
+                }
+                return;
+            }
+        };
+
+        // Abort if the file is empty.
+        if file.metadata().is_ok_and(|m| m.len() == 0) {
+            log::warn!(
+                "Not reading plugin file because it's empty: {}",
+                plugin_path.display()
             );
+            return;
+        }
+
+        // Read the contents of the plugin file
+        let contents = match PluginCacheFile::read_from(&mut file, span) {
+            Ok(contents) => contents,
+            Err(err) => {
+                log::warn!("Failed to read plugin cache file: {err:?}");
+                report_error_new(
+                    engine_state,
+                    &ShellError::GenericError {
+                        error: format!(
+                            "Error while reading plugin cache file: {}",
+                            plugin_path.display()
+                        ),
+                        msg: "plugin path defined here".into(),
+                        span,
+                        help: Some(
+                            "you might try deleting the file and registering all of your \
```

Comment:
Not sure about this `\` since nushell doesn't support line continuation characters like this.

Reply:
this is a Rust syntax feature for wrapping strings

Reply:
ah, right. thx.